util function to ensure positive iota

ddudt opened the pull request:
Adds a function `enforce_positive_iota` to enforce the rotational transform to be positive (without otherwise changing the equilibrium). This is not necessary to enforce, but I find it helpful when comparing equilibria. The implementation is very similar to the existing function `enforce_positive_jacobian`. 

This should have the same effect on ensuring a positive current profile. So maybe we should rename it to `enforce_positive_current` or something more general? 

TODO:

- [x] Properly handle magnetic axis coefficients
- [x] Test when iota profile is assigned
- [x] Test when current profile is assigned

dpanici commented:
test stability quantities as well before/after

ddudt commented:
The code coverage checks might fail, but the only lines that aren't covered are the logic for `EquilibriaFamily` instead of an `Equilibrium`

f0uriest commented on tests/test_compat.py:
```diff
@@ -0,0 +1,162 @@
+"""Tests for compatability functions."""
```
can you also add tests for the directions of the basis vectors? IE, is this flipping the direction of zeta?

f0uriest reviewed (dismissed):
Can you show what the surfaces look like before and after for the following cases:
- axisymmetry
- rotating elipse
- more complicated shape like one of the optimized stellarators

ddudt commented:
> Can you show what the surfaces look like before and after for the following cases:
> 
> * axisymmetry
> * rotating elipse
> * more complicated shape like one of the optimized stellarators

In axisymmetry the boundary does not change at all, we just flip the sign of the iota/current profile. Here is the SOLOVEV example: 
![SOLOVEV](https://github.com/PlasmaControl/DESC/assets/33005725/86da9c1f-635c-46e6-b22b-22c92fc5276f)

For stellarators, the boundary is still the same surface but with the opposite "helicity". 

In the HELIOTRON example (a rotating ellipse), the boundary changes from rotating clockwise to rotating counter-clockwise: 
![HELIOTRON_neg](https://github.com/PlasmaControl/DESC/assets/33005725/a11a50f4-daab-4c53-9bc7-77fa9677aa6a)
![HELIOTRON_pos](https://github.com/PlasmaControl/DESC/assets/33005725/9a281a5a-d0b1-4c1b-8dde-12b8b9fea87c)

In the W7-X example, the positive rotational transform profile is when the boundary is rotating clockwise: 
![W7X_neg](https://github.com/PlasmaControl/DESC/assets/33005725/d6acccea-bcc8-4e78-add3-1db60fd26ab1)
![W7X_pos](https://github.com/PlasmaControl/DESC/assets/33005725/14dfde01-b5f8-4ae5-b7f6-cda1c9b9a8fd)

f0uriest commented:
So isn't this basically just flipping the direction of zeta?

ddudt commented:
Renamed the function to `flip_helicity` and it will always be applied when called, so could be used to make profiles negative if they are already positive. 

Also added more tests. The basis vectors $e_\rho$, $e_\theta$, and $e^\zeta$ are unchanged. $R_\zeta$ and $Z_\zeta$ do flip sign, which changes $e_\zeta$, but these components of the vector are not used to compute $\sqrt{g}$. The QH helicity flips its sign. 

I want to update the QH test to use `"HSX"` instead of the `"precise_QH"` example, so I will wait on #702 for this PR.
